addons/website_slides/tests/test_slide_channel.py

```diff
@@ -53,6 +53,73 @@ def test_archive(self):
             else:
                 self.assertTrue(slide.is_published, "All slides should be unpublished when a channel is archived, except categories")
 
+    def test_channel_partner_next_slide_without_category(self):
+        new_channel = self.env['slide.channel'].create({
+            'name': 'New Channel',
+            'channel_type': 'documentation',
+            'enroll': 'public',
+            'visibility': 'public',
+            'is_published': True,
+        })
+        slide_1 = self.env['slide.slide'].create({
+            'name': 'slide1',
+            'channel_id': new_channel.id,
+            'slide_category': 'document',
+            'is_published': True,
+            'sequence': 1,
+        })
+        slide_2 = self.env['slide.slide'].create({
+            'name': 'slide2',
+            'channel_id': new_channel.id,
+            'slide_category': 'document',
+            'is_published': True,
+            'sequence': 2,
+        })
+
+        new_channel.sudo()._action_add_members(self.user_portal.partner_id)
```

Comment:
This is better, but I think we can simplify.

In a single test:
Create a course with 2 categories and 2 slides per categories.
Add multiple members to the course (at least 2).
Then check the next_slide, complete the slide, check the next one, etc etc.

You can use helper methods and make it look elegant by doing something like:

```py
for member in members:
    for slide in course.slide_ids:
        self.assertEqual(membership.next_slide_id, slide)
        self._mark_as_complete(member, slide)
```

You see the idea?
(Did not implement everything in the code bit above, but it should explain what I mean).
Having multiple members will ensure that the SQL query correctly fetches the next slide based on your own progress and not someone else's.